maat-court-data-api/src/main/java/gov/uk/courtdata/exception/RestControllerAdviser.java

```diff
@@ -1,36 +1,51 @@
 package gov.uk.courtdata.exception;
 
+
 import gov.uk.courtdata.dto.ErrorDTO;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.http.converter.HttpMessageNotReadableException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 
-import java.util.ArrayList;
-import java.util.List;
-
+/**
+ * <code>RestControllerAdviser</code> centralizes all rest controller exceptions.
+ */
+@Slf4j
 @ControllerAdvice
-public class RestControllerAdviser {
-
-
-    @ExceptionHandler({HttpMessageNotReadableException.class, MAATCourtDataException.class})
-    public ResponseEntity<List<ErrorDTO>> handleBadRequest(Exception ex) {
-
-        List<ErrorDTO> dtoList = new ArrayList<>();
-        ErrorDTO.builder().code("error-1")
-                .message("Please check whether request contains all request params.").build();
-        return new ResponseEntity(dtoList,HttpStatus.BAD_REQUEST);
+public class RestControllerAdviser extends ResponseEntityExceptionHandler {
+
+    /**
+     * Redirect all runtime exceptions to internal handler.
+     *
+     * @param ex
+     * @return
+     */
+    @ExceptionHandler({MAATCourtDataException.class})
+    public ResponseEntity<Object> handleRuntimeException(RuntimeException ex, WebRequest request) {
+
+        return handleExceptionInternal(ex,
+                ErrorDTO.builder()
+                        .message(ex.getMessage())
```

Comment:
do we need to set a code value too?

Reply:
correct, we have to set code values once integrated upstream, i had a discussion with Dan from VCD and richard. The messages we send might not be exactly what business wants to see. So when they start integrating the validation endpoints they will come with codes for each fails cases and we send them the code along with a brief description. The VCD will be having a business specific message which also can support english/welsh. Then we have to set the error codes as agreed, we might have a discussion around that time and change accordingly